### PR TITLE
Use Jsoup instead of TagSoup for HTML parsing.

### PR DESCRIPTION
**Pull request by jmartinesp**
Jsoup, being way more flexible, allows us to simplify the parsing *a lot* and do checks we couldn't do before like one which lets us know when extra line breaks between inline nodes and block ones are needed.

### CODE REVIEW

File: platforms/android/library/src/androidTest/java/io/element/android/wysiwyg/inputhandlers/InterceptInputConnectionIntegrationTest.kt [hunk 1]
```diff
@@ -57,7 +58,7 @@ class InterceptInputConnectionIntegrationTest {
             textView.text.dumpSpans(), equalTo(
                 listOf(
                     "hello: android.widget.TextView.ChangeWatcher (0-5) fl=#6553618",
-                    "hello: android.text.style.StyleSpan (0-5) fl=#33",
+                    "hello: android.text.style.StyleSpan (0-5) fl=#17",
```
Comment (jmartinesp) [inline platforms/android/library/src/androidTest/java/io/element/android/wysiwyg/inputhandlers/InterceptInputConnectionIntegrationTest.kt:61]: These changes are caused by using `inSpans`, but they shouldn't affect the functionality of the editor.

File: platforms/android/library/src/main/java/io/element/android/wysiwyg/utils/HtmlToSpansParser.kt [hunk 2]
```diff
@@ -46,464 +42,227 @@ internal class HtmlToSpansParser(
...
+            "i", "em" -> parseInlineFormatting(element, InlineFormat.Italic)
+            "u" -> parseInlineFormatting(element, InlineFormat.Underline)
+            "del" -> parseInlineFormatting(element, InlineFormat.StrikeThrough)
+            "code" -> parseInlineCode(element)
+            "ul", "ol" -> parseChildren(element)
+            "li" -> parseListItem(element)
+            "pre" -> parseCodeBlock(element)
+            "blockquote" -> parseQuote(element)
+            "p" -> parseParagraph(element)
+            "br" -> parseLineBreak(element)
```
Comment (bmarty) [inline platforms/android/library/src/main/java/io/element/android/wysiwyg/utils/HtmlToSpansParser.kt:82]: Maybe add a log for un-handled tag names?

File: platforms/android/library/src/main/java/io/element/android/wysiwyg/utils/HtmlToSpansParser.kt [hunk 3]
```diff
@@ -46,464 +42,227 @@ internal class HtmlToSpansParser(
...
 
-    override fun endElement(uri: String?, localName: String, qName: String?) {
-        handleEndTag(localName)
+    private fun SpannableStringBuilder.parseInlineFormatting(element: Element, inlineFormat: InlineFormat) {
+        val span = when (inlineFormat) {
+            InlineFormat.Bold -> StyleSpan(Typeface.BOLD)
+            InlineFormat.Italic -> StyleSpan(Typeface.ITALIC)
+            InlineFormat.Underline -> UnderlineSpan()
+            InlineFormat.StrikeThrough -> StrikethroughSpan()
+            InlineFormat.InlineCode -> return
```
Comment (bmarty) [inline platforms/android/library/src/main/java/io/element/android/wysiwyg/utils/HtmlToSpansParser.kt:105]: This `return` statement is strange to me.

**Review by bmarty — COMMENTED** [review]
Nice work, thanks!

File: platforms/android/library/src/main/java/io/element/android/wysiwyg/utils/HtmlToSpansParser.kt (hunk 3 above)
Comment (jmartinesp) [reply platforms/android/library/src/main/java/io/element/android/wysiwyg/utils/HtmlToSpansParser.kt:138]: I added some comments to clarify this.

File: platforms/android/library/src/main/java/io/element/android/wysiwyg/utils/HtmlToSpansParser.kt (hunk 2 above)
Comment (jmartinesp) [reply platforms/android/library/src/main/java/io/element/android/wysiwyg/utils/HtmlToSpansParser.kt:97]: Good call! In fact, I added a clean step where we remove any tags and attributes we don't recognise.

File: platforms/android/example-compose/src/main/java/io/element/wysiwyg/compose/MainActivity.kt [hunk 4]
```diff
@@ -81,6 +81,9 @@ class MainActivity : ComponentActivity() {
                 var linkDialogAction by remember { mutableStateOf<LinkAction?>(null) }
                 val coroutineScope = rememberCoroutineScope()
 
+                LaunchedEffect(state.messageHtml) {
+                    Timber.d("Message HTML: '${state.messageHtml}'")
+                }
```
[the remark this answers is not in the document]
Comment (jmartinesp) [reply platforms/android/example-compose/src/main/java/io/element/wysiwyg/compose/MainActivity.kt:86]: Not really, this is in the compose integration example only 😅 .

**Comment by jmartinesp** [comment]
I had to add some extra processing for whitespaces because Jsoup apparently doesn't directly strip it as Tagsoup did, and this could potentially cause index issues in the future.

File: platforms/android/example-compose/src/main/java/io/element/wysiwyg/compose/MainActivity.kt (hunk 4 above)
[the remark this answers is not in the document]
Comment (bmarty) [reply platforms/android/example-compose/src/main/java/io/element/wysiwyg/compose/MainActivity.kt:86]: Ah OK, I read it a bit fast. I am always paranoid regarding what app and libraries are logging out.

**Review by bmarty — APPROVED** [review]
Thanks for the updates!

**Pull request merged**